src/components/LocalesPage/index.js

```diff
@@ -12,25 +14,59 @@ type Props = {
   project: Project
 }
 
-const LocalesPage = ({ project }: Props) => (
-  <SingleProjectLayout>
-    <Wrapper mBottom="xlarge">
-      <Wrapper pBottom="large">
-        <HeaderWrapper>
-          <Texts>
-            <Text size="large">Locales</Text>
-            <Text color="semiDark">{project.name}</Text>
-          </Texts>
-
-          <Button bordered>Add new locale</Button>
-        </HeaderWrapper>
-      </Wrapper>
-
-      {project.locales.map((locale, index) => (
-        <LocaleRow key={locale.code} isEven={index % 2 === 0} locale={locale} />
-      ))}
-    </Wrapper>
-  </SingleProjectLayout>
-)
+class LocalesPage extends React.Component<Props> {
+  dialog = null
+
+  onNewLocaleClick = () => {
+    if (this.dialog) this.dialog.show()
+  }
+
+  render() {
+    const { project } = this.props
+
+    return (
+      <SingleProjectLayout>
+        <Wrapper mBottom="xlarge">
+          <Wrapper pBottom="large">
+            <HeaderWrapper>
+              <Texts>
+                <Text size="large">Locales</Text>
+                <Text color="semiDark">{project.name}</Text>
+              </Texts>
+
+              <Button bordered onClick={this.onNewLocaleClick}>
+                Add new locale
+              </Button>
+            </HeaderWrapper>
+          </Wrapper>
+
+          {project.locales.map((locale, index) => (
+            <LocaleRow
+              key={locale.code}
+              isEven={index % 2 === 0}
+              locale={locale}
+            />
+          ))}
+        </Wrapper>
+        <Dialog
+          id="new-locale-dialog"
+          classNames={{
+            base: 'dialog',
+            document: 'dialog__document',
+            closeButton: 'dialog__close-button',
+            title: 'dialog__title'
+          }}
+          appRoot="#__next"
+          dialogRoot="#dialog-root"
+          dialogRef={dialog => (this.dialog = dialog)}
+          title="Add a new locale"
+          closeButtonContent="× Close"
+        >
+          <NewLocaleModalContainer dialog={this.dialog} project={project} />
```

Comment:
We should pass a callback like `getDialog={this.exposeDialog}` so that `this.dialog` is always up-to-date in child components @rmnbrd (for now it’s undefined on first render)